bot/src/authorisation.ts

```diff
@@ -0,0 +1,73 @@
+import getSigner from 'auctions-core/src/signer';
+import {
+    authorizeCollateral,
+    authorizeWallet,
+    getCollateralAuthorizationStatus,
+    getWalletAuthorizationStatus,
+} from 'auctions-core/src/authorizations';
+import { ETHEREUM_NETWORK, KEEPER_PREAUTHORIZE, WHITELISTED_COLLATERALS } from './variables';
+import { parseCollateralWhitelist } from './whitelist';
+
+export async function authorizeKeeperWallet(authorizeCallback?: () => void) {
+    const signer = await getSigner(ETHEREUM_NETWORK);
+    const walletAddress = await signer.getAddress();
+    const isWalletAuth = await getWalletAuthorizationStatus(ETHEREUM_NETWORK, walletAddress);
+
+    if (!isWalletAuth) {
+        console.info(`keeper: wallet "${walletAddress}" has not been authorized yet. Attempting authorization now...`);
+        const transactionHash = await authorizeWallet(ETHEREUM_NETWORK, walletAddress, false);
+        console.info(`keeper: wallet "${walletAddress}" successfully authorized via "${transactionHash}" transaction`);
+        if (authorizeCallback) {
+            authorizeCallback();
+        }
```

Comment:
Why do you need a callback? I though we're using promises (:

Reply:
I needed it to recall `checkAndParticipateIfPossible`, however I changed it to return 

- `true` if the collateral or wallet was newly authorised
- `false` if the collateral or wallet was previously already authorised

When calling the functions I then check the return value and can then reload the entire function if needed.